routers/api/v1/repo/repo.go

```diff
@@ -500,6 +504,293 @@ func GetByID(ctx *context.APIContext) {
 	ctx.JSON(200, repo.APIFormat(perm.AccessMode))
 }
 
+// Edit edit repository properties
+func Edit(ctx *context.APIContext, opts api.EditRepoOption) {
+	// swagger:operation PATCH /repos/{owner}/{repo} repository repoEdit
+	// ---
+	// summary: Edit a repository's properties. Only fields that are set will be changed.
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo to edit
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo to edit
+	//   type: string
+	//   required: true
+	//   required: true
+	// - name: body
+	//   in: body
+	//   description: "Properties of a repo that you can edit"
+	//   schema:
+	//     "$ref": "#/definitions/EditRepoOption"
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/Repository"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	//   "422":
+	//     "$ref": "#/responses/validationError"
+	owner := ctx.Repo.Owner
+
+	if owner.IsOrganization() && !ctx.User.IsAdmin {
+		isOwner, err := owner.IsOwnedBy(ctx.User.ID)
+		if err != nil {
+			ctx.Error(http.StatusInternalServerError, "IsOwnedBy", err)
+			return
+		} else if !isOwner {
+			ctx.Error(http.StatusForbidden, "", "Given user is not owner of organization.")
+			return
+		}
+	}
```

Comment:
else check if ctx.User == owner ?

Reply:
Isn't this just left to the permission in the routers/api/v1/api.go setup? An owner would be considered and admin, and like Github, admins can change all these settings. https://github.com/go-gitea/gitea/pull/7006/files#diff-2620fdbeeb83ce43692534e6c2c39452R612

Reply:
Honestly, I can't remember why I included that block about the user needing to be a owner if the org if the repo belongs to an org, but know I did that because deleting a repo needs the same permissions.

The same thing is done in the Delete method, but now I wonder if that is necessary for the Edit?

Reply:
Seems like that if-statement is not even needed if the route uses reqAdmin, no?

Reply:
I have resolved that wasn't even needed since I got that from the delete function, and delete requires the user to be an owner of the org, but not to edit, just has to be the admin of the given repo, so I have removed that check.

Admin of the repo is already checked in the routers